adserver/analyzer/backends/naive.py

```diff
@@ -0,0 +1,87 @@
+"""Naive keyword analyzer that is simply based on keyword counts."""
+import collections
+
+from bs4 import BeautifulSoup
+
+from ... import regiontopics as topics
+from .base import BaseAnalyzerBackend
+
+
+class NaiveKeywordAnalyzerBackend(BaseAnalyzerBackend):
+
+    """
+    A very naive keyword analyzer.
+
+    This mimics the results of our ad client.
+    """
+
+    # CSS selectors to select the "main" content of the page
+    # The first of these to match anything is used
+    MAIN_CONTENT_SELECTORS = (
+        "[role='main']",
+        "main",
+        "body",
+    )
+
+    MAX_WORDS_ANALYZED = 1000
```

Comment:
I assume this isn't a performance limit, but that we're just wanting to see what is at the top of the page? We can probably bump this up over time if needed, but probably fine to start with.

Reply:
If anything, I'd reduce it. 1000 words is multiple paragraphs. If it doesn't appear in the first 5-10 paragraphs, it's probably not relevant.

Reply:
Eh, especially with a naive implementation, I feel like all the text is probably useful to analyze, since it will show what is included on the page. If we get a lot smarter about classification we can be pickier, but given the current approach I feel like the more data the better.